README.md

```diff
@@ -66,6 +66,14 @@ object = client.fetch_object_value(flag_key: 'object_value', default_value: JSON
 
 For complete documentation, visit: https://openfeature.dev/docs/category/concepts
 
+### Providers
+
+Providers are allowed to implement an initialize method that is invoked when they are set in the API. In order to avoid conflicting with the Ruby `initialize` method, this method should be named `init`. In addition, it should not accept any parameters and the return value is not used.
+
+Providers are also allowed to implement a `shutdown` method that is invoked on a provider when a new provider is set. This should not accept any parameters and the return value is not used.
```

Comment:
I think the differences between `init` and `initialize`, is more about the specification and working with ruby. If that's the case, this could be targeted at demonstrating how to implement, and then note the differences from the spec at the end.

Towards that end, including a simple example could be useful, ie:

```ruby
class MyProvider
   def init
      # return value not used
   end

  def shutdown
     # return value not used
   end
end
```